backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java

```diff
@@ -0,0 +1,384 @@
+package ai.verta.modeldb.experimentRun.subtypes;
+
+import ai.verta.common.ModelDBResourceEnum;
+import ai.verta.modeldb.CreateExperimentRun;
+import ai.verta.modeldb.ExperimentRun;
+import ai.verta.modeldb.ModelDBConstants;
+import ai.verta.modeldb.common.CommonMessages;
+import ai.verta.modeldb.common.connections.UAC;
+import ai.verta.modeldb.common.exceptions.AlreadyExistsException;
+import ai.verta.modeldb.common.futures.FutureGrpc;
+import ai.verta.modeldb.common.futures.FutureJdbi;
+import ai.verta.modeldb.common.futures.InternalFuture;
+import ai.verta.modeldb.config.Config;
+import ai.verta.modeldb.exceptions.InvalidArgumentException;
+import ai.verta.modeldb.metadata.MetadataServiceImpl;
+import ai.verta.modeldb.utils.ModelDBUtils;
+import ai.verta.modeldb.utils.TrialUtils;
+import ai.verta.modeldb.versioning.EnvironmentBlob;
+import ai.verta.modeldb.versioning.PythonEnvironmentBlob;
+import ai.verta.modeldb.versioning.PythonRequirementEnvironmentBlob;
+import ai.verta.uac.*;
+import java.util.*;
+import java.util.concurrent.Executor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class CreateExperimentRunHandler {
+
+  private static Logger LOGGER = LogManager.getLogger(CreateExperimentRunHandler.class);
+
+  private final Executor executor;
+  private final FutureJdbi jdbi;
+  private final UAC uac;
+  private final Config config = Config.getInstance();
+
+  private final AttributeHandler attributeHandler;
+  private final KeyValueHandler hyperparametersHandler;
+  private final KeyValueHandler metricsHandler;
+  private final ObservationHandler observationHandler;
+  private final TagsHandler tagsHandler;
+  private final ArtifactHandlerBase artifactHandler;
+  private final FeatureHandler featureHandler;
+
+  public CreateExperimentRunHandler(Executor executor, FutureJdbi jdbi, UAC uac) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.uac = uac;
+
+    attributeHandler = new AttributeHandler(executor, jdbi, "ExperimentRunEntity");
+    hyperparametersHandler =
+        new KeyValueHandler(executor, jdbi, "hyperparameters", "ExperimentRunEntity");
+    metricsHandler = new KeyValueHandler(executor, jdbi, "metrics", "ExperimentRunEntity");
+    observationHandler = new ObservationHandler(executor, jdbi);
+    tagsHandler = new TagsHandler(executor, jdbi, "ExperimentRunEntity");
+    artifactHandler = new ArtifactHandlerBase(executor, jdbi, "artifacts", "ExperimentRunEntity");
+    featureHandler = new FeatureHandler(executor, jdbi, "ExperimentRunEntity");
```

Comment:
Don't create new via copy and paste. Re-use the existing ones. Follow the pattern already shown in other handlers.

Reply:
i do not follow you on this. Could you explain what is wrong here??

Reply:
I have used whatever we have and if it is not anywhere in your or mine PR then i am creating new one.

Reply:
This looks like a copy and paste of this: https://github.com/VertaAI/modeldb/blob/master/backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java#L42

Reply:
Ok, got it so mean i have to use those in parameters for create handler. ok, i will do that. thanks for clarification.

Reply:
PR: https://github.com/VertaAI/modeldb/pull/2196/files